crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1152,6 +1158,56 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
             )));
         }
 
+        // Check whether this version covers all supported platforms.
+        if !dist.implied_markers().is_true() {
+            for marker in [
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "linux".to_string(),
+                }),
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "win32".to_string(),
+                }),
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "darwin".to_string(),
+                }),
```

Comment:
Note that we don't enforce these if the current environment is disjoint. So, like, if a user _needs_ to resolve a requirement that only has Windows wheels, they can set `environments = ["sys_platform == 'win32'"]`.